### ISSUE TITLE
[BeAM] fails to build

### ISSUE THREAD

**Issue by diversys** [haikuports/haikuports#2162]
http://vmpkg.haiku-os.org/master/x86_gcc2/logviewer.html?buildruns/782/builds/27031.log

```
LexC++ generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.c 
Cc generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.o 
/sources/Beam-280e3862f83bedc05d683455aed79be112a680f8/src-filter-addons/src-sieve/src-libSieve/addr-lex.l:29: addr.h: No such file or directory
In file included from /boot/system/develop/headers/posix/unistd.h:10,
                from /boot/system/develop/headers/bsd/unistd.h:9,
                from /sources/Beam-280e3862f83bedc05d683455aed79be112a680f8/generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.c:810:
/boot/system/develop/headers/posix/stdint.h:64: warning: `INT16_MIN' redefined
/sources/Beam-280e3862f83bedc05d683455aed79be112a680f8/generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.c:321: warning: this is the location of the previous definition
/boot/system/develop/headers/posix/stdint.h:69: warning: `INT32_MIN' redefined
/sources/Beam-280e3862f83bedc05d683455aed79be112a680f8/generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.c:324: warning: this is the location of the previous definition
/boot/system/develop/headers/posix/stdint.h:132: warning: `SIZE_MAX' redefined
/sources/Beam-280e3862f83bedc05d683455aed79be112a680f8/generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.c:346: warning: this is the location of the previous definition

gcc -c -o generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.o -DBEAM_FOR_BONE -DBEAM_FOR_HAIKU -Wall -Wno-multichar -Wmissing-prototypes -Wpointer-arith -Wcast-align -Wsign-compare -DENABLE_REGEX -Wno-conversion -Wno-unused -Wno-sign-compare -Wno-uninitialized -O3 -fstrict-aliasing  -Isrc-filter-addons/src-sieve/src-libSieve -Igenerated/objects-haiku/src-filter-addons/src-sieve/src-libSieve -Isrc-bmBase -I- -I/boot/system/include generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.c

...failed Cc generated/objects-haiku/src-filter-addons/src-sieve/src-libSieve/addr-lex.o ...
```

**Comment by diversys** [comment]
Weirdly enough it builds fine locally.

**Comment by pulkomandy** [comment]
This depends on the version of flex being used. Please compare local and remote versions and see which one is broken.

**Comment by diversys** [comment]
In both cases it's flex-2.6.4-1-x86_gcc2.hpkg. 

```
--- vmpkg	2018-01-28 11:25:56.021757952 +0300
+++ local	2018-01-28 11:26:04.984875008 +0300
@@ -15,10 +15,9 @@
 grep-2.24-1-x86_gcc2.hpkg
 gutenprint-5.2.13-2-x86_gcc2.hpkg
 gzip-1.8-2-x86_gcc2.hpkg
-haiku-r1~alpha4_pm_hrev51727-1-x86_gcc2.hpkg
-haiku_devel-r1~alpha4_pm_hrev51727-1-x86_gcc2.hpkg
+haiku-r1~alpha4_pm_hrev51655-1-x86_gcc2.hpkg
+haiku_devel-r1~alpha4_pm_hrev51655-1-x86_gcc2.hpkg
 icu-57.1-2-x86_gcc2.hpkg
-ilmbase-2.2.0-6-x86_gcc2.hpkg
 jam-2.5_2012_10_12-5-x86_gcc2.hpkg
 jasper-2.0.14-1-x86_gcc2.hpkg
 jpeg-9b-5-x86_gcc2.hpkg
@@ -37,7 +36,6 @@
 ncurses6-6.0-3-x86_gcc2.hpkg
 noto-20170920-3-any.hpkg
 noto_sans_cjk_jp-1.004-2-any.hpkg
-openexr-2.2.0-6-x86_gcc2.hpkg
 openssl-1.0.2n-1-x86_gcc2.hpkg
 openssl_devel-1.0.2n-1-x86_gcc2.hpkg
 readline-7.0.3-1-x86_gcc2.hpkg

```

**Comment by pulkomandy** [comment]
Fixed in 2072d369f2938.
